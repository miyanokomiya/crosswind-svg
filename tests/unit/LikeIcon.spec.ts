import { mount } from "@vue/test-utils";
import Target from "../../src/components/LikeIcon.vue";

describe("LikeIcon", () => {
  it("liked:false", () => {
    const wrapper = mount(Target, {
      propsData: { liked: false }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("liked:true", () => {
    const wrapper = mount(Target, {
      propsData: { liked: true }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
