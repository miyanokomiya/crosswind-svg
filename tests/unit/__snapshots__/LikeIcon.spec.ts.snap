// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeIcon liked:false 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor; user-select: none;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
  >
    <path
      d="M 32 48 Q 10 32 20 22 Q 26 16 32 26 Q 38 16 44 22 Q 54 32 32 48"
      fill="none"
      stroke-linecap="round"
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 50% 50%;"
    />
     
    <path
      d="M 32 48 Q 10 32 20 22 Q 26 16 32 26 Q 38 16 44 22 Q 54 32 32 48"
      fill=""
      stroke-linecap="round"
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 50% 50%; transform: scale(0);"
    />
  </g>
</svg>
`;

exports[`LikeIcon liked:true 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor; user-select: none;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
  >
    <path
      d="M 32 48 Q 10 32 20 22 Q 26 16 32 26 Q 38 16 44 22 Q 54 32 32 48"
      fill="none"
      stroke-linecap="round"
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 50% 50%; transform: scale(0);"
    />
     
    <path
      d="M 32 48 Q 10 32 20 22 Q 26 16 32 26 Q 38 16 44 22 Q 54 32 32 48"
      fill=""
      stroke-linecap="round"
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 50% 50%;"
    />
  </g>
</svg>
`;
